src/framework/components/anim/system.js

```diff
@@ -89,6 +89,22 @@ class AnimComponentSystem extends ComponentSystem {
     }
 
     cloneComponent(entity, clone) {
+        let masks;
```

Comment:
looks good, but this not take place even when the rootBone is specified? It can point to the root of the hierarchy in which case it's the same case?
I also wonder if we should in debug mode validate those paths against the hierarchy? Or does it take place when binding takes place? To make sure it's compatible skeleton.

Reply:
Yeah, I was thinking that the rootBone would be null in that case (it is on the editor side). That should probably be replicated in the engine logic (assigning the components entity as the rootBone should result in a null rootBone).

The debugging would need to occur in the binder.

Reply:
I think it would not be null if the user assigned root entity to it?

Reply:
Perhaps we should null the rootBone property if the components entity is assigned to it? To mirror the editor logic of entity assignments? Or we do a check for the component's entity as the rootBone here.

Reply:
I'd prefer we don't change the rootBone assignment .. if user assigns something, it should stay assigned, even if it's the same as the default case.

Reply:
Ok i've updated this check to include the current entity.